geoh5py/objects/block_model.py

```diff
@@ -170,21 +177,18 @@ def u_cell_delimiters(self) -> np.ndarray | None:
         :obj:`numpy.array` of :obj:`float`:
         Nodal offsets along the u-axis relative to the origin.
         """
-        if (
-            getattr(self, "_u_cell_delimiters", None) is None
-        ) and self.existing_h5_entity:
-            delimiters = self.workspace.fetch_delimiters(self.uid)
-            self._u_cell_delimiters = delimiters[0]
-            self._v_cell_delimiters = delimiters[1]
-            self._z_cell_delimiters = delimiters[2]
+        if (getattr(self, "_u_cell_delimiters", None) is None) and self.on_file:
+            self._u_cell_delimiters = self.workspace.fetch_array_attribute(
+                self.uid, "u_cell_delimiters"
+            )
```

Comment:
They are all done independently since they are technically all seperate attributes. It is just nicer to handle all the array type attribute the same way... all just becomes a loop over keys

Reply:
yup